pg_analytics/src/hooks/process.rs

```diff
@@ -37,6 +38,10 @@ pub fn process_utility(
         completion_tag: *mut pg_sys::QueryCompletion,
     ) -> HookResult<()>,
 ) -> Result<(), ParadeError> {
+    if needs_commit()? {
+        task::block_on(commit_writer())?;
+    }
+
```

Comment:
Are we doing the same thing here as in `pg_bm25`, where we're committing mid-transaction? E.g. if the transaction fails after this commit, then we are incorrectly not rolling back to the state before the transaction?

It's fine if that's the case... but since we have to think about how to solve it for `pg_bm25`, I just want to track if the same problem exists here.

Reply:
Yea, if the transaction fails but `commit_writer` succeeds then the row will be commited. delta-rs keeps track of table versions, would adding an abort callback and rolling back the delta table version solve this issue?

Reply:
Yes, I think that would solve the issue, as long as you can track the table version from the beginning of the _transaction_, and not just from the last time that `commit_writer` was called in the current transaction.

I understand if you'd rather do that in a follow up PR, I'll stamp this one if you want to get it in.

Reply:
Awesome! i'll open up an issue for it